deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/PodParentHandler.java

```diff
@@ -108,5 +120,55 @@ protected Object prepareScalePatch(ScaleOperation scaleOp, int scaleValue, int c
                 .build();
         return K8sResourcePatchUtil.getJsonPatch(exisiting, scale, V1Scale.class);
     }
+    
+    public PodParent getPodParent(ApiClient apiClient, String selector, boolean label, String namespace) throws HyscaleException {
+        List<PodParent> podParentList = getPodParentsList(apiClient, selector, label, namespace);
+        if (podParentList != null && !podParentList.isEmpty()) {
+            return podParentList.get(0);
+        }
+        return null;
+    }
+    
+    public List<PodParent> getPodParentsList(ApiClient apiClient, String selector, boolean label, String namespace)
+            throws HyscaleException {
+        List<PodParent> podParents = new ArrayList<PodParent>();
+
+        List podParentResource = null;
+        try {
+            podParentResource = getBySelector(apiClient, selector, label, namespace);
+        } catch (HyscaleException e) {
+            if (!e.getHyscaleErrorCode().equals(DeployerErrorCodes.RESOURCE_NOT_FOUND)) {
+                throw e;
+            }
+        }
+        if (podParentResource != null) {
+            podParentResource.stream().forEach(each -> {
+                PodParent podParent = new PodParent(getKind(), each);
+                podParents.add(podParent);
+            });
+        }
+        return podParents;
+    }
+    
+    public List<PodParent> getParentsForAllNamespaces(ApiClient apiClient, String selector, boolean label) throws HyscaleException {
```

Comment:
This can still be removed from here since we moved it to PodParentProvider right ?

Reply:
This is in line with getParents since both are using different api calls.

Reply:
getAllParents makes more sense else it is confusing

Reply:
Will remove these methods from PodParentHandler and only have an implementation in PodParentProvider which will have 4 methods
hasPodParent
getPodParent - return single podParent.
getPodParents - takes namespace as argument
getAllPodParents
Is this approach better?

Reply:
getPodParent - return single podParent.
getPodParents - takes namespace as argument

Not getting the essence of these methods !

Reply:
getPodParent is to get a single Pod Parent where as getPodParents is to get all the Pod Parents based on the selector. getPodParent will stop at the first pod parent it discovers.